Thanks for the tip. I added a test using `launch_testing`. This seems to work best.

I checked that:
```
colcon test --packages-select rqt_controller_manager --event-handlers console_direct+
```
passes with this commit, and fails with:
```
[...]
[rqt_controller_manager-1] AttributeError: type object 'Qt' has no attribute 'CustomContextMenu'
[rqt_controller_manager-1] 
[rqt_controller_manager-1] 
[rqt_controller_manager-1] thread '<unnamed>' (762017) panicked at /home/buildfarm/.cargo/git/checkouts/zenoh-9c599d5ef3e0480e/2687c51/commons/zenoh-runtime/src/lib.rs:154:21:
[rqt_controller_manager-1] The Thread Local Storage inside Tokio is destroyed. This might happen when Zenoh API is called at process exit, e.g. in the atexit handler. Calling the Zenoh API at process exit is not supported and should be avoided.
[rqt_controller_manager-1] note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[ERROR] [rqt_controller_manager-1]: process has died [pid 762017, exit code -6, cmd '[WS]/install/rqt_controller_manager/lib/rqt_controller_manager/rqt_controller_manager --ros-args -r __node:=rqt_controller_manager_node'].
----------------------------- Captured stderr call -----------------------------
test_node_start (test_run_launch.TestFixture.test_node_start) ... FAIL

======================================================================
FAIL: test_node_start (test_run_launch.TestFixture.test_node_start)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "[WS]/src/ros2_control/rqt_controller_manager/test/test_run_launch.py", line 40, in test_node_start
    assert node_name in self.node.get_node_names()
AssertionError: assert 'rqt_controller_manager_node' in ['_ros2cli_daemon_0_a3b603a242054a28be47c1554cef0efc', 'test_node']
 +  where ['_ros2cli_daemon_0_a3b603a242054a28be47c1554cef0efc', 'test_node'] = get_node_names()
 +    where get_node_names = <rclpy.node.Node object at 0x72b0f402c2f0>.get_node_names
 +      where <rclpy.node.Node object at 0x72b0f402c2f0> = <test_run_launch.TestFixture testMethod=test_node_start>.node

----------------------------------------------------------------------
Ran 1 test in 2.044s

FAILED (failures=1)

----------------------------------------------------------------------
Ran 0 tests in 0.000s

NO TESTS RAN
- generated xml file: [WS]/build/rqt_controller_manager/pytest.xml -
=========================== short test summary info ============================
FAILED test/test_run_launch.py::test_run_launch - test_run_launch.TestFixture.test_node_start failed
============================== 1 failed in 2.53s ===============================
Finished <<< rqt_controller_manager [3.88s]     [ with test failures ]
```
when reverting the commit.

--- a/rqt_controller_manager/package.xml
+++ b/rqt_controller_manager/package.xml
@@ -27,6 +27,9 @@
   <exec_depend>rqt_gui</exec_depend>
   <exec_depend>rqt_gui_py</exec_depend>
 
+  <test_depend>launch_testing</test_depend>
+  <test_depend>launch_testing_ros</test_depend>
+
   <export>
     <build_type>ament_python</build_type>
     <rqt_gui plugin="${prefix}/plugin.xml"/>

--- a/rqt_controller_manager/rqt_controller_manager/controller_manager.py
+++ b/rqt_controller_manager/rqt_controller_manager/controller_manager.py
@@ -308,14 +308,14 @@ class ControllerManager(Plugin):
         # Show context menu
         menu = QMenu(self._widget.ctrl_table_view)
         action_toggle_auto_resize = menu.addAction("Toggle Auto-Resize")
-        action = menu.exec_(ctrl_header.mapToGlobal(pos))
+        action = menu.exec(ctrl_header.mapToGlobal(pos))
 
         # Evaluate user action
         if action is action_toggle_auto_resize:
-            if ctrl_header.resizeMode(0) == QHeaderView.ResizeToContents:
-                ctrl_header.setSectionResizeMode(QHeaderView.Interactive)
+            if ctrl_header.sectionResizeMode(0) == QHeaderView.ResizeMode.ResizeToContents:
+                ctrl_header.setSectionResizeMode(QHeaderView.ResizeMode.Interactive)
             else:
-                ctrl_header.setSectionResizeMode(QHeaderView.ResizeToContents)
+                ctrl_header.setSectionResizeMode(QHeaderView.ResizeMode.ResizeToContents)
 
     def _update_hw_components(self):
         if not self._cm_name:
@@ -373,7 +373,7 @@ class ControllerManager(Plugin):
             action_configure = menu.addAction(self._icons["inactive"], "Configure")
             action_spawn = menu.addAction(self._icons["active"], "Configure and Activate")
 
-        action = menu.exec_(self._widget.hw_table_view.mapToGlobal(pos))
+        action = menu.exec(self._widget.hw_table_view.mapToGlobal(pos))
 
         # Evaluate user action
         if hw_component.state.label == "active":
@@ -431,14 +431,14 @@ class ControllerManager(Plugin):
         # Show context menu
         menu = QMenu(self._widget.hw_table_view)
         action_toggle_auto_resize = menu.addAction("Toggle Auto-Resize")
-        action = menu.exec_(hw_header.mapToGlobal(pos))
+        action = menu.exec(hw_header.mapToGlobal(pos))
 
         # Evaluate user action
         if action is action_toggle_auto_resize:
-            if hw_header.resizeMode(0) == QHeaderView.ResizeToContents:
-                hw_header.setSectionResizeMode(QHeaderView.Interactive)
+            if hw_header.sectionResizeMode(0) == QHeaderView.ResizeMode.ResizeToContents:
+                hw_header.setSectionResizeMode(QHeaderView.ResizeMode.Interactive)
             else:
-                hw_header.setSectionResizeMode(QHeaderView.ResizeToContents)
+                hw_header.setSectionResizeMode(QHeaderView.ResizeMode.ResizeToContents)
 
     def _activate_controller(self, name):
         self._switch_controllers([name], [])
@@ -604,11 +604,11 @@ class FontDelegate(QStyledItemDelegate):
     def paint(self, painter, option, index):
         if not index.parent().isValid():
             # Root level
-            option.font.setWeight(QFont.Bold)
+            option.font.setWeight(QFont.Weight.Bold)
         if index.parent().isValid() and not index.parent().parent().isValid():
             # Hardware interface level
             option.font.setItalic(True)
-            option.font.setWeight(QFont.Bold)
+            option.font.setWeight(QFont.Weight.Bold)
         QStyledItemDelegate.paint(self, painter, option, index)
 
 

--- a/rqt_controller_manager/test/test_run_launch.py
+++ b/rqt_controller_manager/test/test_run_launch.py
@@ -1,0 +1,57 @@
+# Copyright (c) 2026 Christian Rauch
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import unittest
+
+import launch
+import launch_ros.actions
+import launch_testing.actions
+import launch_testing.markers
+import pytest
+import rclpy
+from rclpy.node import Node
+
+node_name = "rqt_controller_manager_node"
+
+
+@pytest.mark.launch_test
+@launch_testing.markers.keep_alive
+def generate_test_description():
+    return launch.LaunchDescription(
+        [
+            launch_ros.actions.Node(
+                executable="rqt_controller_manager",
+                package="rqt_controller_manager",
+                name=node_name,
+                additional_env={"QT_QPA_PLATFORM": "offscreen"},
+            ),
+            launch_testing.actions.ReadyToTest(),
+        ]
+    )
+
+
+class TestFixture(unittest.TestCase):
+
+    def setUp(self):
+        rclpy.init()
+        self.node = Node("test_node")
+
+    def tearDown(self):
+        self.node.destroy_node()
+        rclpy.shutdown()
+
+    def test_node_start(self, proc_output):
+        time.sleep(2)
+        assert node_name in self.node.get_node_names()

